blocks/init/src/Blocks/components/heading/heading.php

```diff
@@ -26,19 +26,17 @@
 
 $headingClass = Components::classnames([
 	Components::selector($componentClass, $componentClass),
+	Components::selector($blockClass, $blockClass),
 	Components::selector($blockClass, $blockClass, $selectorClass),
 	Components::selector($additionalClass, $additionalClass),
 ]);
 
 $headingLevel = $headingLevel ? "h{$headingLevel}" : 'h2';
 
-$unique = Components::getUnique();
+$unique = $attributes['uniqueWrapperId'] ?? Components::getUnique();
+echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Comments about the unique function and putting the CSS output inside the elements are valid for every case, so I won't repeat them on every file 😄